### ISSUE TITLE
Exception while getting json() of DELETE responses

### ISSUE THREAD

**Issue by husayt** [117/alpaca#52]
**Description**
There are a few methods in Alpaca api which are done through `DELETE` call. It is not standard for DELETE calls to return result, but here they do. The problem is when we try to do `fetch().json()` they throw an exception. This library currently wraps that exception and returns `{}`, but then we are missing the original result. 

**Expected**
To return the received response from function.

**Reproduction**
Steps we can take to reproduce the bug:
For exampe, trying to cancel an order with `cancelOrder(id)` you don't know what was the result.

**Additional**
This is due to fetch handling `DELETE` messages (at least in a browser). Need to find a proper way of handling it

**Comment by 117** [comment]
Good catch.

**Comment by 117** [comment]
Do you think the status should be returned or the entire response?

```typescript
// if json parse fails we default to status return
.then(async (resp) => (await resp.json().catch(() => false)) || reject(resp.status))
```

**Comment by husayt** [comment]
This is not really solving the problem. Exception handling is for unseen problems. When we know that certain call is causing an error, we should better find a way to do it properly without exceptions, rather than hiding the error.

Especially for this kind of critical api, it is important to preserve and pass on everything we receive. The call is to cancel an order, what if order was not cancelled and we return ok? Consequences could be absolutely damning.

**Comment by husayt** [comment]
Actually for this particular request here is what documentation say:

> [DELETE] Cancel an order
> DELETE/v2/orders/{order_id}
> Attempts to cancel an open order. If the order is no longer cancelable (example: status="filled"), the server will respond with status 422, and reject the request. Upon acceptance of the cancel request, it returns status 204.

So we shouldn't even try to do `json()` here, as there is no body returned in response. 
I will come up with suggestion, also that method has mix of async and promises syntax. Will convert it all to async for better readability

**Comment by husayt** [comment]
fix is in #53
